test/Microsoft.ML.TestFramework/DataPipe/TestDataPipe.cs

```diff
@@ -490,6 +492,157 @@ public void SavePipeCountSelectWithSparse()
+                        uint u4 = 0;
+                        ulong u8 = 0;
+
+                        while (c.MoveNext())
+                        {
+                            getI1(ref i1);
+                            getI2(ref i2);
+                            getI4(ref i4);
+                            getI8(ref i8);
+                            getU1(ref u1);
+                            getU2(ref u2);
+                            getU4(ref u4);
+                            getU8(ref u8);
+
+                            // Signed to smaller signed.
+                            VerifyMatch(i8, i1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(i4, i1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(i2, i1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(i8, i2, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(i4, i2, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(i8, i4, Conversions.Instance.Convert, Conversions.Instance.Convert);
+
+                            // Unsigned to smaller unsigned.
+                            VerifyMatch(u8, u1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(u4, u1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(u2, u1, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(u8, u2, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(u4, u2, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                            VerifyMatch(u8, u4, Conversions.Instance.Convert, Conversions.Instance.Convert);
+                        }
+                    }
+                }, logCurs: true);
+
+            Done();
+        }
+
+        private bool VerifyMatch<TSrc, TDst>(TSrc src, TDst dst, ValueMapper<TSrc, TDst> conv, ValueMapper<TDst, TSrc> convBack)
+            where TSrc : struct
+            where TDst : struct
+        {
+            TDst v = default(TDst);
+            conv(ref src, ref v);
+            if (EqualityComparer<TDst>.Default.Equals(dst, v))
+                return true;
+            TSrc vSrc = default;
+            convBack(ref v, ref vSrc);
+            if (EqualityComparer<TDst>.Default.Equals(dst, default(TDst)) && !EqualityComparer<TSrc>.Default.Equals(src, vSrc))
+                return true;
+            Fail($"Values different values in VerifyMatch<{typeof(TSrc).Name}, {typeof(TDst).Name}>: converted from {typeof(TSrc).Name} to {typeof(TDst).Name}: {v}. Parsed from text: {dst}");
+            return false;
+        }
+
+        [Fact(Skip = "Fails until issue #1342 is resolved.")]
```

Comment:
This test doesnt work when SelectColumns is used? There maybe an underlying transform that is calling SelectColumns.

Reply:
Yes, I think the NgramHash transform uses temporary columns and drops them at the end.

---
In reply to: [228275145](https://github.com/dotnet/machinelearning/pull/1375#discussion_r228275145) [](ancestors = 228275145)